conf/nginx.conf.template

```diff
@@ -500,10 +500,74 @@ http {
       proxy_pass __S3_ENDPOINT_PROTO__$aws_tgt_bucket/$encoded_key;
     }
 
+    # Check an object without downloading it.
+    #
+    # For some reason that I did not understand, HEAD is failing with 403 againt GCS
+    # In order to have a fast way to check if an object exist without a possibly expensive
+    # listing or download, we can do a GET on a short range without caching
+    #
+    # Will return 416 if the object exists (the range header is invalid on purpose)
+    # Will return 404 if the object does not exist
+    #
+    location ~ ^/check_raw/([^/]+/.*[^/])$ {
```

Comment:
Basically, this is a poor man's HEAD implementation with an invalid range that will trigger 416 only if the object exists, otherwise it will return 404.